### PR TITLE
Migrated to f-strings for .\.nuget\create_nuget.py  #4136

### PR DESCRIPTION
**Pull request by coldkillerr**
Related to issue #4136

### CODE REVIEW

File: .nuget/create_nuget.py [hunk 1]
```diff
@@ -64,7 +64,7 @@
     </ItemGroup>
     </Project>
     """
-    target_str = r"""
+    target_str = f"""
```
Comment (jameslamb) [inline .nuget/create_nuget.py:67]:
```suggestion
    target_str = r"""
```

This needs to be a raw string and does not rely on any Python template. Please change this back.

File: .nuget/create_nuget.py [hunk 2]
```diff
@@ -45,8 +45,8 @@
         <file src="runtimes\**" target="runtimes"/>
         </files>
     </package>
-    """ % (version, datetime.datetime.now().year)
-    prop_str = r"""
+    """ 
+    prop_str = f"""
```
Comment (jameslamb) [inline .nuget/create_nuget.py:49]:
```suggestion
    prop_str = r"""
```

This needs to be a raw string and does not rely on any Python template. Please change this back.

File: .nuget/create_nuget.py [hunk 3]
```diff
@@ -25,11 +25,11 @@
     copy_file(os.path.join(source, "lib_lightgbm.dll"), os.path.join(windows_folder_path, "lib_lightgbm.dll"))
     copy_file(os.path.join(source, "lightgbm.exe"), os.path.join(windows_folder_path, "lightgbm.exe"))
     version = open(os.path.join(current_dir, os.path.pardir, 'VERSION.txt')).read().strip().replace('rc', '-rc')
-    nuget_str = r"""<?xml version="1.0"?>
+    nuget_str = f"""<?xml version="1.0"?>
     <package xmlns="http://schemas.microsoft.com/packaging/2013/05/nuspec.xsd">
     <metadata>
         <id>LightGBM</id>
-        <version>%s</version>
+        <version>{version, datetime.datetime.now().year}</version>
```
Comment (jameslamb) [inline .nuget/create_nuget.py:32]: This fix is not quite correct. The `%s` should be replaced with `version` and the `%d` on line 39 should be replaced with `datetime.datetime.now().year`.

**Review by jameslamb — CHANGES_REQUESTED** [review]
Thank you for taking the time to contribute to LightGBM!

In the future, when you make a pull request based on a previous conversation, please link to that conversation. For example, in the description you could write "Part of #4136".

Please revert the changes you've made to `docs/conf.py`. There is already an open pull request addressing the change to f-strings for that file (#4147).

Please see the additional requested changes below.

File: .nuget/create_nuget.py (hunk 2 above)
Comment (coldkillerr) [reply .nuget/create_nuget.py]: I have added the requested changes to create_nuget.py.

File: .nuget/create_nuget.py (hunk 3 above)
Comment (coldkillerr) [reply .nuget/create_nuget.py]: fixed it

**Comment by coldkillerr** [comment]
Do I have to make any more changes?

File: .nuget/create_nuget.py [hunk 4]
```diff
@@ -25,18 +25,18 @@
     copy_file(os.path.join(source, "lib_lightgbm.dll"), os.path.join(windows_folder_path, "lib_lightgbm.dll"))
     copy_file(os.path.join(source, "lightgbm.exe"), os.path.join(windows_folder_path, "lightgbm.exe"))
     version = open(os.path.join(current_dir, os.path.pardir, 'VERSION.txt')).read().strip().replace('rc', '-rc')
-    nuget_str = r"""<?xml version="1.0"?>
+    nuget_str = f"""<?xml version="1.0"?>
```
Comment (StrikerRUS) [inline .nuget/create_nuget.py:28]:
```suggestion
    nuget_str = rf"""<?xml version="1.0"?>
```

File: .nuget/create_nuget.py [hunk 5]
```diff
@@ -25,18 +25,18 @@
...
-        <version>%s</version>
+        <version>{version}</version>
         <authors>Guolin Ke</authors>
         <owners>Guolin Ke</owners>
         <licenseUrl>https://github.com/microsoft/LightGBM/blob/master/LICENSE</licenseUrl>
         <projectUrl>https://github.com/microsoft/LightGBM</projectUrl>
         <requireLicenseAcceptance>false</requireLicenseAcceptance>
         <description>A fast, distributed, high performance gradient boosting framework</description>
-        <copyright>Copyright %d @ Microsoft</copyright>
+        <copyright>Copyright {datetime.datetime.now().year}@ Microsoft</copyright>
```
Comment (StrikerRUS) [inline .nuget/create_nuget.py:39]:
```suggestion
        <copyright>Copyright {datetime.datetime.now().year} @ Microsoft</copyright>
```

**Review by StrikerRUS — CHANGES_REQUESTED** [review]

**Review by jameslamb — APPROVED** [review]
changes look good to me, thanks very much!

File: .nuget/create_nuget.py [hunk 6]
```diff
@@ -45,7 +45,7 @@
         <file src="runtimes\**" target="runtimes"/>
         </files>
     </package>
-    """ % (version, datetime.datetime.now().year)
+    """
```
Comment (StrikerRUS) [inline .nuget/create_nuget.py:48]:
```suggestion
    """
```

**Review by StrikerRUS — APPROVED** [review]
LGTM except one trailing whitespace symbol.

**Comment by jameslamb** [comment]
@coldkillerr when you address https://github.com/microsoft/LightGBM/pull/4162#discussion_r606841237, can you please also merge in the latest `master`?

Sorry for the inconvenience, but we recently merged in a change that fixes some issues in this project's continuous integration (#4161).

**Comment by jameslamb** [comment]
I'm going to use admin rights to merge this one, since it's fairly self-contained and only failing the CUDA builds (which were temporarily removed in #4167 ).

**Comment by jameslamb** [comment]
Thanks for the help @coldkillerr !

**Pull request merged**